pkg/solana/txm/txm.go

```diff
@@ -505,6 +514,82 @@ func (txm *Txm) handleErrorSignatureStatus(sig solanaGo.Signature, status *rpc.S
 	}
 }
 
+// handleReorg detects and manages transaction state regressions (re-orgs) for a given signature.
+//
+// A re-org occurs when the on-chain state of a signature regresses:
+// - Confirmed -> Processed || Not Found
+// - Processed -> Not Found
+//
+// This function determines if the signature’s state regression impacts the overall transaction state and, if so, takes appropriate action:
+//   - For regressions from "Confirmed", our in memory layer is updated and the tx is rebroadcasted with a new hash restarting the retry/bumping cycle.
+//   - For regressions from "Processed", the existing retry/bumping cycle is still running, so no immediate action is needed. We only update our in-memory state to Broadcasted.
+//     Future rebroadcasts, will be handled by the TxExpirationRebroadcast logic (if enabled) when the transaction expires.
+func (txm *Txm) handleReorg(ctx context.Context, client client.ReaderWriter, sig solanaGo.Signature, status *rpc.SignatureStatusesResult) {
+	// Retrieve the last known status of the transaction associated with this signature from the in-memory layer.
+	txInfo, err := txm.txs.GetSignatureInfo(sig)
+	if err != nil {
+		txm.lggr.Errorw("failed to get signature info when checking for potential re-orgs", "signature", sig, "error", err)
+		return
+	}
+
+	// Check if the sig status has regressed to indicate a re-org.
+	// A regression is identified when the state transitions as follows:
+	// - Confirmed -> Processed || Broadcasted || Not Found
+	// - Processed -> Broadcasted || Not Found
+	currentTxState := convertStatus(status)
+	if regressionType, isRegressed := isStatusRegression(txInfo.state, currentTxState); isRegressed {
```

Comment:
If we don't need `regressionType` anymore based on my other comment, I think we can just use `TxHasReorg` for this and eliminate `GetSignatureInfo` and `UpdateSignatureStatus` to minimize the number of storage methods we use.
1. We could rework `TxHasReorg` to take in a sig and a current state and return the id and bool. That way we can do the `isStatusRegression` on the storage side and we can drop `GetSignatureInfo`
2. We could also update `TxHasReorg` to fetch the id from the sig map to fetch the tx. Since we'd have access to the current sig and its new state, I don't think we'd have to call `UpdateSignatureStatus`. We'd be dropping all old sigs anyways.